### DISCUSSION TITLE
Reject emails for disabled alias (e.g. SMTP error "Recipient not found")

### DISCUSSION THREAD

**Discussion by jnv** [simple-login/app#478] [Feature Requests]
Hi, first of all, thank you for Simplelogin! I am a happy paying subscriber and I am glad this service exists.

Previously I used just a standard domain catch-all and Simplelogin allowed me to cut off compromised aliases. However, in some cases I receive somewhat legitimate e-mails to some of my disabled aliases (e.g. cold contacts from recruiters to my Git commit address). Currently when the alias is disabled, the received emails just disappear without any bounce back to the sender.

So I was thinking whether you would consider an optional feature to allow disabled alias to act as non-existing recipient. From my limited understanding of SMTP, the protocol allows to communicate that through error code [550 5.1.1](https://mailtrap.io/blog/550-5-1-1-rejected-fix/). I don't propose to enable this by default for all disabled aliases, but perhaps a checkbox or additional option beside "enabled" and "disabled" to opt-in for this feature.

I don't have much insight into spammers behavior, so maybe it won't discourage them from sending further emails. But legitimate senders would be informed they are trying to use an invalid address.

**Comment by nguyenkims** [comment]
The reason SimpleLogin returns 250 status for disabled alias is that so we can re-enable an alias and start receiving emails again. I wonder for your usecase, this ticket can be useful? https://github.com/simple-login/app/discussions/406

This way you don't have to disable a whole alias but only blocks some senders or just allow some senders to continue sending emails to the alias?

**Reply by jnv** [reply]
Thanks Kim! I guess blocking senders would mostly solve my problem because so far I see spammers mostly reusing the same domains.

I see that if I attempt to send an email to removed alias, I get proper 550 error from the server. So technically this feature is already present.

I think my issue is complicated by the fact that the alias is hosted on my domain and I have also `Catch All` enabled, so even if I delete the alias, it gets recreated once anyone sends an email to that alias. So I guess removing the Catch All and deleting that alias is a possible solution here. Unfortunately that's not a possibility for me since I didn't keep a track of what aliases I used in the past.

So, just thinking out loud. I am not sure if removed aliases keep some kind of "tombstones" so they don't get reassigned to other users, but if this is the case, perhaps these "tombstones" can override the Catch All functionality on custom domains? By removing the alias I would accept that I am effectively giving up any possibility of recreating it in the future.

**Reply by nguyenkims** [reply]
When you delete an alias, all emails sent to it have the 550 error as this alias can't be recreated easily. Actually, only deleted aliases created with your domain can be restored, the ones created with built in domains can't (for security reason).

**Reply by WhamboMPS** [reply]
This is an FYI regarding spammer behavior for @jnv and @nguyenkims.  I was a (legitimate!) email marketer by trade and so am more familiar with the behavior of this group than I should be. Here are some thoughts:

1. Differentiate spam from "bacn". Spam is something you don't want and never asked for. Bacn -- something between spam and real meat (!) -- is email you asked to receive but which may not be very valuable. Examples of bacn include newsletters from a e-commerce business, notifications from a job posting site when you aren't looking for work, etc.
2. It's easy to stop receiving bacn. Senders of bacn are the types of businesses that I used to work for. They are _very concerned_ about their ability to get the legitimate email they send to their users. If they get too many complaints of "This is spam" to the company that manages the endpoint mailbox infrastructure (e.g. Google for Gmail, AOL for Yahoo! Mail, etc.) they will get **_ALL_** their marketing email blocked for a time which costs big bucks. Senders of bacn are typically more than happy to remove those that don't want their email from their mailing list - just unsubscribe.
3. True spammers either guess email addresses (a form of an indiscriminate directory attack) or they obtain actual email addresses through a fraudulent or illegal process. Think signups that are bogus or thefts of registration databases.
4. True spammers worry about the reputation of their IP address as much as legitimate commercial mailers do! If a spammer blasts ten million pieces of true spam out of a single IP address and/or email address users will hit the "This is spam" button in droves and that sending IP address and email address will be blocked by the receiving infrastructure. This is the reason why "spam bots" exist: they allow the spammer to send their messages from a swarm of different computers, each of which are unlikely to send enough spam in a short enough period of time to trip the "reputation" based filters that block spam at the receiving infrastructure level.
5. True spammers share email addresses amongst each other. If one spammer has a compromised destination email email address there is a good chance that scores, hundreds or even thousands of other spammers have it as well. Once an email gets loose in the wild there is no way to put the cap back on that genie's bottle. This is the main rationale for services like SimpleLogin.
6. Spammers primarily send email from computers belonging to others. Their true marginal cost of sending a single email is virtually zero. As a result the spammer has no financial incentive to keep a "clean" list. The computational cost of handling bounces is incurred by the bot net member's computer that is sending the email so the spammer doesn't care about bounces, either. You cannot inflict enough pain on the spammer to get them to change their behavior. The only thing you can do is turn off (or delete) the email address that is receiving the spam messages.

Hope this helps!

**Reply by jnv** [reply]
Thanks for the clarification @WhamboMPS but my concern isn't about bacn, it's about legitimate cold contact emails which are mixed up with spam. So this isn't bacn because I haven't asked for these emails, but they may be still interesting for me.

I use an email alias for [git commits](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git). Since this email is pushed to public repositories, it is publicly accessible in git history. This is where it gets picked by various contact harvesters and either [leaked](https://haveibeenpwned.com/), or directly harvested by spammers.

Now there are legitimate uses for this, many recruiters and salespeople use services like Amazinghiring to contact leads through GitHub. I don't mind receiving these emails. Sadly, the same email alias is being frequently hit by spam. So my original intent with this feature request was to communicate to the sender "hey, you can't reach me this way" – a legitimate sender may find another way to contact me and spammers just won't care.

I know  GitHub offers a [noreply email address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-on-github) but in this case this email alias is already out there and changing my commit email address now won't change much. My best bet is to ask those data brokers like Amazinghiring to remove my personal details, and either accept the spam coming to that alias, or forego mildly interesting cold contacts by disabling the alias. I chose the latter option.

**Comment by fazy** [comment]
I too would like to be able to choose between rejecting (bouncing) emails or silently dropping them.

When an alias is compromised and gets all kinds of blatant spam/malware, dropping makes sense.

When an alias is getting abused by recruiters (new ones pop up every day, and even Facebook recruiters don't comply with GDPR), bouncing the email will send them the message that they need to stop (and if they use something like MailChimp it'll show in their stats). Filtering based on sender doesn't work in this case.

Also, I still want to auto-create aliases; I just want more control once a specific alias is being abused.

**Reply by nguyenkims** [reply]
@fazy I see the use case now. Do you think this should be a global setting or a setting per alias?

**Reply by jnv** [reply]
I suggest a per-alias setting, perhaps you may want to disable some aliases only temporarily.

**Comment by fazy** [comment]
Yes, I think per alias would be better because the profile of spam can be different for different compromised addresses. It might not even be compromised, it might be that a supposedly legitimate organisation doesn't act on the unsubscribe request.

**Comment by fazy** [comment]
I'm not sure if this has been implemented yet, so I tried a workaround and stumbled upon another:

Firstly, used regex and the much appreciated auto create alias rules feature. Something like `^prefix(?!foo$|bar$).+$` where foo and bar are addresses I want to bounce. It matched "prefixsomething" (creating the alias) but not "prefixfoo" or "prefixbar" (emails bounce).

However when testing this, I also created an address like this:

prefixbaz@example.com

and then deleted it, at which point it went into the domain trash. My intention had been to add `baz` to the regex and check that it also bounces, but I wanted to see what happens when it's in the domain trash, and it bounced anyway.

So I guess there are two "workarounds":

1. Use a regex (with extreme care and testing, and bearing in mind it will get unmanageable after a few excluded addresses)
2. Delete the domain and don't empty the trash

Is the trash behaviour an intended implementation of this feature request, or is there going to be another way coming?

Also, does the trash auto-empty after a while?

**Reply by nguyenkims** [reply]
@fazy I'm not sure to fully understand the question ... When an alias is moved to the custom domain trash, it can't receive emails. The same error happens when an alias can't be auto-created (i.e. it doesn't match any auto-create rule).

> Also, does the trash auto-empty after a while?
 
No the trash doesn't auto empty.

**Comment by fazy** [comment]
Thanks for the quick reply. :) 

I guess I was trying to work out if the Trash feature is supposed to implement this feature request. From the OP:

> So I was thinking whether you would consider an optional feature to allow disabled alias to act as non-existing recipient.

It seems we can get that behaviour by deleting the alias and leaving it in the trash. However that has other side effects, such as removing the history/stats, and cannot be undone.

So the original feature request is still useful, however I think for _some_ use cases, we can get the behaviour of rejecting emails. I must admit I didn't understand it fully but I think I've got it now.

Oh, and my Regex trick will only be useful if you empty the trash by mistake and don't want to wait for aliases to be re-created before blocking them again.
